cmd/dex/config.go

```diff
@@ -182,15 +183,36 @@ type OAuth2 struct {
 
 // Web is the config format for the HTTP server.
 type Web struct {
-	HTTP           string   `json:"http"`
-	HTTPS          string   `json:"https"`
-	Headers        Headers  `json:"headers"`
-	TLSCert        string   `json:"tlsCert"`
-	TLSKey         string   `json:"tlsKey"`
-	TLSMinVersion  string   `json:"tlsMinVersion"`
-	TLSMaxVersion  string   `json:"tlsMaxVersion"`
-	AllowedOrigins []string `json:"allowedOrigins"`
-	AllowedHeaders []string `json:"allowedHeaders"`
+	HTTP           string         `json:"http"`
+	HTTPS          string         `json:"https"`
+	Headers        Headers        `json:"headers"`
+	TLSCert        string         `json:"tlsCert"`
+	TLSKey         string         `json:"tlsKey"`
+	TLSMinVersion  string         `json:"tlsMinVersion"`
+	TLSMaxVersion  string         `json:"tlsMaxVersion"`
+	AllowedOrigins []string       `json:"allowedOrigins"`
+	AllowedHeaders []string       `json:"allowedHeaders"`
+	ClientRemoteIP ClientRemoteIP `json:"clientRemoteIP"`
+}
+
+type ClientRemoteIP struct {
+	Header            string   `json:"header"`
+	TrustedProxyCIDRs []string `json:"trustedProxyCIDRs"`
+}
+
+func (cr *ClientRemoteIP) ToParsedCIDRs() (string, []netip.Prefix, error) {
```

Comment:
Why return the header here?

Reply:
At first, I thought this method could prepare everything for IP address detection. If you think this is odd, I can just remove it, if you don't mind. 

Fixed.

Reply:
I think I like the latest version better. If there is need for some reason to pack all that into a single method, I'd call it something like `prepareClientIPDetection`.

But as I said, I like the latest version.